sdk/tables/azure-data-tables/migration_guide.md

```diff
@@ -0,0 +1,298 @@
+
+### New features
+
+We have a variety of new features available in the new Azure Data Tables library:
+* Ability to submit a batch of operations with the `TableClient.submit_transaction` method
+* Separates Table and Entity level operations into two clients, `TableServiceClient` and `TableClient`
+* Adds the ability to query tables and list entities explicitly.
+
+## Important changes
+
+### Package names and namespaces
+
+The package name has been changed from `azure-cosmosdb-table` to `azure-data-tables`. This package can be used to access and manipulate data stored in a CosmosDB Table account or an Azure Storage Tables account.
+
+### Client hierarchy and constructors
+
+In the interest of simplicity, there are only two clients, `TableServiceClient` for account-level interactions and `TableClient` for table-level interactions. This is in contrast to the single `ServiceClient` used for interacting with the account level operations and entity level operations. The two clients will simplify the method calls by not requiring a Table name to be included on each service call.
+
+### Authenticating Clients
+
+In `azure-cosmosdb-table`:
+```python
+import os
+from azure.cosmosdb.table import TableService
+
+client = TableService(account_name=os.environ["ACCOUNT_URI"], account_key=os.environ["ACCOUNT_KEY"])
+```
+
+In `azure-data-tables`:
+```python
+import os
+from azure.core.credentials import AzureNamedKeyCredential
+from azure.data.tables import TableClient, TableServiceClient
+
+key = os.environ["TABLES_ACCOUNT_KEY"]
+name = os.environ["TABLES_ACCOUNT_NAME"]
+url = os.environ["TABLES_ACCOUNT_URL"]
+
+credential = AzureNamedKeyCredential(key=key, name=name)
+
+table_client = TableClient(account_url, "tablename", credential=credential)
+
+table_service_client = TableServiceClient(account_url, credential=credential)
+```
+
+
+### Table Level Scenarios
+
+#### Create and Delete
+
+In `azure-cosmosdb-table`:
+```python
+from azure.cosmosdb.table import TableService
+client = TableService(...)
+table_name = "tableName"
+# create_table returns True if a new table was created, False if the table already exists
+created = client.create_table(table_name)
+
+# Delete
+table_name = "deleteTableName"
+# delete_table returns True if a table was deleted, False if the table does not exist
+client.delete_table(table_name)
+```
+
+In `azure-data-tables` the `TableClient` or `TableServiceClient` can be used to create a new table or delete an existing table:
+```python
+from azure.data.tables import TableServiceClient, TableClient
+
+# You can create a table from the TableServiceClient or TableClient
+service_client = TableServiceClient.from_connection_string(conn_str)
+table_client = service_client.create_table("tableName")
+service_client.delete_table("tableName")
+
+# The create_table_if_not_exists is an alternative to create_table
+table_client = TableClient.from_connection_string(conn_str, table_name="tableName")
+table_client.create_table_if_not_exists()
```

Comment:
This is incorrect. `create_table_if_not_exists` is only on the `TableServiceClient`